CDTDatastore/CDTReplicator/CDTAbstractReplication.h

```diff
@@ -146,6 +146,16 @@ typedef NS_ENUM(NSInteger, CDTReplicationErrors) {
  * @return an initialsed instance of CDTAbstractReplication.
  */
 - (instancetype)initWithUsername:(nullable NSString*)username password:(nullable NSString*)password;
+
+/**
+ * Initalises the abstract replication, using an IAM API key to authenticate.
+ * See https://console.bluemix.net/docs/services/Cloudant/guides/iam.html#ibm-cloud-identity-and-access-management
+ * for more information about IAM.
+ * @param iamApiKey The IAM API key.
+ * @return an initialsed instance of CDTAbstractReplication.
+ */
+- (instancetype)initWithIamApiKey:(NSString *)iamApiKey;
```

Comment:
There is a mix of `Iam` and `IAM` in different places. There are more `IAM` than `Iam` so I think the smaller change is to uppercase it everywhere. I think this is also acceptable by convention where well known acronyms are permitted. I guess it is debatable whether IAM is "well known", whilst certainly not as well known as `URL` or `HTTP` I think it could be considered well known in this context with documentation etc.